packages/opentelemetry-sdk-trace-base/test/common/Span.test.ts

```diff
@@ -486,6 +488,38 @@ describe('Span', () => {
         });
       });
 
+      describe('when span "attributeCountLimit" set to the default value and general "attributeCountLimit" option defined', () => {
```

Comment:
The test name says "set to the default value" but it could just as well be any other value, right?

Reply:
No, for this use case it must be set to the default value

Reply:
Can you explain why?
The relevant part in the code is checking against `null`:

```js
  if (spanLimits.attributeCountLimit == null && userConfig.generalLimits?.attributeCountLimit != null) {
    spanLimits.attributeCountLimit = userConfig.generalLimits.attributeCountLimit;
  }
```

My understanding is that the test works the way it does because `spanLimits.attributeCountLimit != null`

Reply:
You can find the test that covers regular use case [here](https://github.com/open-telemetry/opentelemetry-js/blob/42b8409de2e0c495216f1d2e22b7ffe82ee530d5/packages/opentelemetry-sdk-trace-base/test/common/Span.test.ts#L458) . It's already there but it couldn't test/recognize the bug solved by this PR.  
The new test covers the case when OTEL_SPAN_ATTRIBUTE_COUNT_LIMIT is explicitly set to its default value (as wirtten in the bug description), so only for the bug being solved here and that's why it states "set to the default value".